lambda/MAS.Tests/UnitTests/MailServiceTests.cs

```diff
@@ -0,0 +1,34 @@
+﻿using MAS.Tests.Infrastructure;
+using System.Threading.Tasks;
+using Xunit;
+using MAS.Services;
+using MailChimp.Net.Interfaces;
+using Moq;
+using MailChimp.Net.Models;
+using MailChimp.Net.Core;
+using Shouldly;
+
+namespace MAS.Tests.UnitTests
+{
+    public class MailServiceTests : TestBase
+    {
+        [Fact]
+        public void SendToMailChimp()
+        {
+            //Arrange
+            var mockMailChimpManager = new Mock<IMailChimpManager>();
+            mockMailChimpManager.Setup(x => x.Campaigns.AddAsync(It.IsAny<Campaign>())).ReturnsAsync(new Campaign() { Id = "1234" });
+            mockMailChimpManager.Setup(x => x.Content.AddOrUpdateAsync(It.IsAny<string>(), It.IsAny<ContentRequest>()));
+            mockMailChimpManager.Setup(x => x.Campaigns.SendAsync(It.IsAny<string>()));
+
+            var mailService = new MailService(mockMailChimpManager.Object);
+
+            //Act
+            var ex = Record.ExceptionAsync(async () => await mailService.CreateAndSendCampaignAsync("Test Subject", "Preview Text", "Body Text"));
+
+            //Assert
+            ex.Exception.ShouldBe(null);
+            ex.Status.ShouldBe(TaskStatus.RanToCompletion);
```

Comment:
We know the task ran to completion but how we do we know that mail chip actually send the message through? We're checking to see if CreateAndSendCampaignAsync worked should also be checking that the subject preview text and body text were sent as part of the message? We are inputting them but that doesn't guarantee the code will send them all through.

Reply:
I like how pragmatic you are Imran! Testing the behavior itself rather than the feedback mechanism seems like a solid way to test as it helps to avoid false positives.

Reply:
Do we need to get the campaignId and then get the status to be sure that it sent?